StudyHub/StudyHub.BLL/Services/SubjectService.cs

```diff
@@ -121,4 +95,66 @@ public async Task<SubjectDTO> UpdateSubjectAsync(Guid userId, Guid subjectId, Up
 
         return _mapper.Map<SubjectDTO>(entity);
     }
+
+    public async Task<List<StudentDTO>> AddStudentsToSubjectAsync(
+        Guid subjectId,
+        Guid teacherId,
+        StudentsToSubjectRequest request)
+    {
+        var subject = await _subjectRepository.FirstOrDefaultAsync(s => s.Id == subjectId)
+            ?? throw new NotFoundException("Subject not found with the specified Id");
+
+        if (subject.TeacherId != teacherId)
+            throw new RestrictedAccessException("You are not the owner and do not have permission to perform this action.");
+
+        foreach (var email in request.Emails)
+        {
+            var user = await _userManager.FindByEmailAsync(email)
+                ?? throw new NotFoundException($"Student not found with the specified email: {email}");
```

Comment:
Let's image, we have 3 emails which we wanted to add, first pass correctly, second will no found in user manager, what happen to third one? And also, we partly have succesfull adding, but we will have an Exception. Think, needed some speculate here